mds/views.py

```diff
@@ -116,7 +117,14 @@ class DeviceViewSet(
     UpdateOnlyModelMixin,
     MultiSerializerViewSet,
 ):
-    queryset = models.Device.objects.all()
+    queryset = models.Device.objects.all().prefetch_related(
+        Prefetch(
+            "telemetries",
+            queryset=models.Telemetry.objects.order_by("-timestamp"),
+            to_attr="ordered_telemetries",
+        )
```

Comment:
This is better as you do not get the N+1 problem anymore, still you fetch all telemetries for each vehicle, which seems a lot just to get the last telemetry date. Subquery might do a better jog than prefetch, cf. https://docs.djangoproject.com/en/2.1/ref/models/expressions/#subquery-expressions